format/spec.md

```diff
@@ -318,7 +318,7 @@ Partition field IDs must be reused if an existing partition spec contains an equ
 | **`truncate[W]`** | Value truncated to width `W` (see below)                     | `int`, `long`, `decimal`, `string`                                                                        | Source type |
 | **`year`**        | Extract a date or timestamp year, as years from 1970         | `date`, `timestamp`, `timestamptz`, `timestamp_ns`, `timestamptz_ns`                                      | `int`       |
 | **`month`**       | Extract a date or timestamp month, as months from 1970-01-01 | `date`, `timestamp`, `timestamptz`, `timestamp_ns`, `timestamptz_ns`                                      | `int`       |
-| **`day`**         | Extract a date or timestamp day, as days from 1970-01-01     | `date`, `timestamp`, `timestamptz`, `timestamp_ns`, `timestamptz_ns`                                      | `int`       |
+| **`day`**         | Extract a date or timestamp day, as days from 1970-01-01     | `date`, `timestamp`, `timestamptz`, `timestamp_ns`, `timestamptz_ns`                                      | `date`      |
```

Comment:
A date is also an integer, as it represents the number of days since 1-1-1970. I think that `int` is correct here, similar to `month`, `hour`, and others.

Reply:
While the underlying type of date is an int, shouldn't it should still be documented to be date, as it's expected that metadata tables produce `YYYY-mm-dd` formatted values for days (I gathered this from #279)?  As it's written now, it's surprising to me to see non-integer values when I look at partition metadata.

Reply:
Agree with Fokko, this is a description of the transform which takes a date or timestamp and produces and int. If you are implementing this transform and return the implementation would be incorrect. Now if you implement this function and display the output as a different type (like the metadata table) that's fine.